### DISCUSSION TITLE
[FEAT] Support for Astro

### DISCUSSION THREAD

**Discussion by SalahAdDin** [barvian/number-flow#82] [Ideas]
Your component is amazing for improving the UX/UI of our websites.

We want you to support Astro too. 

How could we help you with it?

**Comment by maxiCalderonBuono** [comment]
Hello! I'm also looking for Astro support! It would be amazing since more and more of us are using Astro in our daily work, in my case, alongside Svelte.

**Reply by barvian** [reply]
Hi, thanks for the feedback. Would the [Svelte version](https://number-flow.barvian.me/svelte) work for you in this case?

**Reply by ruizdiazever** [reply]
Hi, I'm using number-flow with shadcn-svelte (next) and Svelte 5 on Astro 5 and it works great, maybe something to improve is to add a small guide to install the package on the specific targets (Svelte, React, etc.)

**Reply by maxiCalderonBuono** [reply]
Still same error for me, I'm using it in a plain Svelte 5 component and then use it on an Astro 5 component. Even using the directive client:only='svelte' I keep on getting Error hydrating: TypeError matchMedia is not a function. I've also tried different versions and still the same.

**Reply by barvian** [reply]
@maxiCalderonBuono Hi, could you [file an issue](https://github.com/barvian/number-flow/issues/new?template=bug_report.yml) for this? I haven't seen that error before 🤔

**Reply by ruizdiazever** [reply]
I haved the same problem in Svelte 5 without shadcn-svelte D:

**Comment by barvian** [comment]
Hi, thanks for the discussion. For clarification, do you mean a vanilla JS astro component? i.e. no framework

**Comment by barvian** [comment]
Hi, I've released a vanilla JS version of NumberFlow with some SSR instructions for Astro:

https://number-flow.barvian.me/vanilla

**Reply by ruizdiazever** [reply]
Bravo, thanks!

**Reply by SalahAdDin** [reply]
Thanks man!

**Reply by maxiCalderonBuono** [reply]
Thank you!. Much appreciated.
